src/connector/src/sink/redis.rs

```diff
@@ -38,35 +90,296 @@ impl Sink for RedisSink {
     type Writer = RedisSinkWriter;
 
     async fn new_writer(&self, _writer_env: SinkWriterParam) -> Result<Self::Writer> {
-        todo!()
+        Ok(RedisSinkWriter::new(
+            self.config.clone(),
+            self.schema.clone(),
+            self.pk_indices.clone(),
+            self.is_append_only,
+        )?)
     }
 
     async fn validate(&self, _client: Option<ConnectorClient>) -> Result<()> {
-        todo!()
+        let client = self.config.common.build_client()?;
+        client.get_connection()?;
+        Ok(())
     }
 }
 
-pub struct RedisSinkWriter;
+pub struct RedisSinkWriter {
+    // connection to redis, one per executor
+    conn: Option<Connection>,
+    // the command pipeline for write-commit
+    pipe: Pipeline,
+    kv_formatter: Option<(String, String)>,
+    epoch: u64,
+    schema: Schema,
+    is_append_only: bool,
+    pk_indices: Vec<usize>,
+}
+impl RedisSinkWriter {
+    pub fn new(
+        config: RedisConfig,
+        schema: Schema,
+        pk_indices: Vec<usize>,
+        is_append_only: bool,
+    ) -> Result<Self> {
+        let client = config.common.build_client()?;
+        let conn = Some(client.get_connection()?);
+        let pipe = redis::pipe();
+        let kv_formatter = match (config.common.key_format, config.common.value_format) {
+            (Some(key_format), Some(value_format)) => Some((key_format, value_format)),
+            _ => None,
```

Comment:
This means user needs to specifiy both key_format and value_format. Specifying only one of them will not work and the default is used for both. If this is the intention, can we add a check in `validate` to make sure specifying only one format is an error?